### PR TITLE
Added a prototype for Level

### PR DESCRIPTION
**Pull request by harry-7**
In reference to #90. Even though it does not fullfill all requirements it provides a basic idea.
@niccokunzmann See the changes and comment on the status.

### CODE REVIEW

File: javascript/levels.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
+function Level(home, backgroundLayers, obstaclesLayer){
+	this._backgrounds = backgroundLayers;
+	this.obstacleLayer = obstaclesLayer;
```
Comment (niccokunzmann) [inline javascript/levels.js]: I think, we can have several obstacle layers

File: javascript/levels.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+	this._obstacles = createObstacles(obstaclesLayer);
+	this._name = name;
```
Comment (niccokunzmann) [inline javascript/levels.js]: where does name com from?

**Comment by harry-7** [comment]
@niccokunzmann I removed the typos and added the multiple obstacle thing as well.

File: javascript/levels.js [hunk 3]
```diff
@@ -59,21 +69,62 @@ function newLevel(name,backgroundLayers,obstaclesLayer){
 }
 
 function AlllevelObjects(){
```
Comment (niccokunzmann) [inline javascript/levels.js]: Should this be executed several times? Can you remove this function and make the spec a global variable?

File: javascript/levels.js [hunk 4]
```diff
@@ -59,21 +69,62 @@ function newLevel(name,backgroundLayers,obstaclesLayer){
...
+			"obstacles":[
+				{
+					"layer": "cactus",
+					"velocity": 10,
+				},
+			],
+
+			"backgrounds":[
+				{
+	                "layer" : "DayAndNight",
```
Comment (niccokunzmann) [inline javascript/levels.js]: Could you format these lines?

Also, the usage of tabs and spaces should be consistent throughout the project. If not, we mix it and the source code appears depending on the editor settings.
What would you like to use?

File: javascript/levels.js [hunk 5]
```diff
@@ -59,21 +69,62 @@ function newLevel(name,backgroundLayers,obstaclesLayer){
...
+				}
+			],
+		},
+
 	};
 
-	for(level in levels){
-		all_levels[level] = newLevel(level,backgrounds[level],obstacles[level]);
+
+	for( level in all_level_specifications ){
```
Comment (niccokunzmann) [inline javascript/levels.js]: I really like these lines. This is short, understandable and says the levels know what to do with the specification.

File: javascript/levels.js [hunk 6]
```diff
@@ -1,6 +1,6 @@
-function Level(home, backgroundLayers, obstaclesLayer){
-	this._backgrounds = backgroundLayers;
-	this.obstacleLayer = obstaclesLayer;
+function Level(name, backgrounds, obstacles){
+	this._backgrounds = backgrounds;
+	this.obstacles = obstacles;
 	this._obstacles = createObstacles(obstaclesLayer);
```
Comment (niccokunzmann) [inline javascript/levels.js]: What does this line do?

File: javascript/levels.js (hunk 4 above)
Comment (harry-7) [inline javascript/levels.js]: I am fine with anything. I ll go with what is already there in the project.

File: javascript/levels.js (hunk 3 above)
Comment (harry-7) [inline javascript/levels.js]: Everytime you start a game this function should be executed.

File: javascript/levels.js [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+function getAllLevels(){
+	return all_levels;
+}
+
+function newLevel(name,backgroundLayers,obstaclesLayer){
+	return new Level(name,backgroundLayers,obstaclesLayer);
+}
+
+
+function AlllevelObjects(){
```
Comment (niccokunzmann) [inline javascript/levels.js]: I expect [getAllLevels](https://github.com/fossasia/flappy-svg/pull/117/files#diff-ab2fd88104fdeff5658e06ebd4b2434fR116) to return the same levels all the time.
Thus, I expect AlllevelObjects() to be called only once.

There are several ways to meet this expectation, which I see:
- Run this function in this file when it is loaded and make it "private"
- Remove the function around the for-loop and execute the for-loop when the file is loaded.

What is your perspective on this?

**Comment by harry-7** [comment]
I would prefer the second one . running it with window.onload. I made necessary changes.

**Comment by niccokunzmann** [comment]
I agree with what is written there, we can merge :)

**Pull request merged**